src/station_gui.cpp

```diff
@@ -531,6 +514,49 @@ class CompanyStationsWindow : public Window
 			SetDParam(0, this->window_number);
 			SetDParam(1, this->vscroll->GetCount());
 		}
+
+		if (widget == WID_STL_CARGODROPDOWN) {
+			if (this->filter.cargoes == 0) {
+				SetDParam(0, this->filter.include_empty ? STR_STATION_LIST_CARGO_FILTER_ONLY_EMPTY : STR_STATION_LIST_CARGO_FILTER_NO_CARGO_TYPES);
+			} else if (this->filter.cargoes == _cargo_mask) {
+				SetDParam(0, this->filter.include_empty ? STR_STATION_LIST_CARGO_FILTER_ALL_AND_EMPTY : STR_CARGO_TYPE_FILTER_ALL);
+			} else if (CountBits(this->filter.cargoes) == 1 && !this->filter.include_empty) {
+				SetDParam(0, CargoSpec::Get(FindFirstBit(this->filter.cargoes))->name);
+			} else {
+				SetDParam(0, STR_STATION_LIST_CARGO_FILTER_MULTIPLE);
+			}
+		}
+	}
+
+	DropDownList BuildCargoDropDownList(bool expanded) const
+	{
+		/* Define a custom item consisting of check mark, count string, icon and name string. */
+		using DropDownListCargoItem = DropDownCheck<DropDownString<DropDownListIconItem, FS_SMALL, true>>;
+
+		DropDownList list;
+		list.push_back(std::make_unique<DropDownListStringItem>(STR_STATION_LIST_CARGO_FILTER_SELECT_ALL, CF_SELECT_ALL));
+		list.push_back(std::make_unique<DropDownListDividerItem>(-1));
+
+		int count = this->stations_per_cargo_type[NUM_CARGO];
+		SetDParam(0, count);
+		std::string count_str = GetString(STR_JUST_INT);
```

Comment:
Seems like the complicated version of `std::string count_str = fmt::format("{}", this->stations_per_cargo_type[NUM_CARGO]);`. Is it really necessary to go through our string formatting system here?